src/ripple/app/tx/impl/CreateTicket.cpp

```diff
@@ -23,105 +23,135 @@
 #include <ripple/protocol/Feature.h>
 #include <ripple/protocol/Indexes.h>
 #include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/st.h>
 
 namespace ripple {
 
 NotTEC
 CreateTicket::preflight (PreflightContext const& ctx)
 {
-    if (! ctx.rules.enabled(featureTickets))
+    if (! ctx.rules.enabled (featureTicketBatch))
         return temDISABLED;
 
     if (ctx.tx.getFlags() & tfUniversalMask)
         return temINVALID_FLAG;
 
-    auto const ret = preflight1 (ctx);
-    if (!isTesSuccess (ret))
-        return ret;
-
-    if (ctx.tx.isFieldPresent (sfExpiration))
     {
-        if (ctx.tx.getFieldU32 (sfExpiration) == 0)
-        {
-            JLOG(ctx.j.warn()) <<
-                "Malformed transaction: bad expiration";
-            return temBAD_EXPIRATION;
-        }
+        // In C++17: put "count" into if's init-statement, lose outer braces.
+        std::uint32_t const count {ctx.tx[sfCount]};
+        if (count < minValidCount || count > maxValidCount)
+            return temINVALID_COUNT;
+    }
+    {
+        // In C++17: put "ret" into if's init-statement, lose outer braces.
+        NotTEC const ret {preflight1 (ctx)};
+        if (!isTesSuccess (ret))
+            return ret;
     }
-
     return preflight2 (ctx);
 }
 
+TER
+CreateTicket::preclaim(PreclaimContext const& ctx)
+{
+    auto const id = ctx.tx[sfAccount];
+    auto const sleAccountRoot = ctx.view.read(keylet::account(id));
+    if (! sleAccountRoot)
+        return terNO_ACCOUNT;
+
+    // Make sure the TicketCreate would not cause the account to own
+    // too many tickets.
+    auto const optTicketCount {(*sleAccountRoot)[~sfTicketCount]};
+    std::uint32_t const curTicketCount {optTicketCount ? *optTicketCount : 0u};
+    std::uint32_t const addedTickets {ctx.tx[sfCount]};
```

Comment:
Haven't checked that this will work with copy initialization, but I suspect they will.